src/components/scenes/Loans/LoanDetailsScene.tsx

```diff
@@ -139,6 +117,90 @@ export const LoanDetailsScene = (props: Props) => {
     } else return null
   }
 
+  // #region Loan Action Cards
+
+  const collateralPositions = React.useMemo(() => collaterals.filter(collateral => !zeroString(collateral.nativeAmount)), [collaterals])
+  const debtPositions = React.useMemo(() => debts.filter(debt => !zeroString(debt.nativeAmount)), [debts])
+
+  const actionCards = React.useMemo(() => {
```

Comment:
It seems appropriate to use a render function here: `renderActionCards` instead of memoizing React elements.

I think as soon as you have this pattern:

```tsx
const foo = useMemo(() => <JSX />, [a, b, c])
```

Then you really should just have a new component:

```tsx
const Foo = ({ a, b, c}) => {
  return <JSX />
}
```

React will only render if the a, b, and c props change, just like the a, b, and c deps in the hook.

But with all that said, we have this pattern of "render functions" which are really just impure-components and get around the overhead of creating a new component and file for the component. Maybe we should consider "local components" which can exist in the same file as the component which uses it, and it isn't exported from the file. I'd prefer that pattern over "render functions" or memoized JSX.